brave/src/main/java/brave/internal/baggage/ExtraHandler.java

```diff
@@ -89,8 +90,13 @@ protected ExtraHandler(Object initialState) {
    * @see TraceContext.Extractor#extract(Object)
    */
   // This operates on the builder directly to allow us to hide newExtra and potential bugs
-  public final E provisionExtra(TraceContextOrSamplingFlags.Builder builder, Object request) {
+  public final <R> E provisionExtra(
+      TraceContextOrSamplingFlags.Builder builder,
+      Getter<R, String> getter,
```

Comment:
this is needed for secondary sampling to set localSampled conditionally. A future PR will integrate that... just I needed to make sure only String keys are possible before proceeding.